src/gcp/secretManager.ts

```diff
@@ -63,123 +94,131 @@ export async function secretExists(projectId: string, name: string): Promise<boo
 }
 
 export function parseSecretResourceName(resourceName: string): Secret {
-  const nameTokens = resourceName.split("/");
+  const tokens = resourceName.match(SECRET_NAME_REGEX);
+  if (tokens == null) {
+    throw new FirebaseError(`Invalid secret resource name [${resourceName}].`);
+  }
   return {
-    projectId: nameTokens[1],
-    name: nameTokens[3],
+    projectId: tokens[1],
```

Comment:
If you're going to go through the work of a bunch of non-capture groups and captures by index I wonder if it's worth having named captures instead.